__tests__/plugins/mattermost/__unit__/MattermostClient.test.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+import { SuperAgentRequest } from 'superagent';
+import { MattermostClient } from '../../../../src/plugins/mattermost/MattermostClient';
+import { MattermostMiddleware } from '../../../../src/plugins/mattermost/MattermostMiddleware';
+import { IMattermostOption } from '../../../../src/types';
+
+describe('MMClient unit', () => {
+  // Expect the post and get reqs to succeed (no error) under various https+tlsCert combinations
+  it('empty, blank, or null tlsOptions', () => {
+    const snapAgent = (agent: SuperAgentRequest, testName: string) => {
+      // agent._ca tracks the loaded cert authority, and is not dumped as part of normal object
+      expect((agent as any)._ca).toMatchSnapshot(testName);
+    };
```

Comment:
smart way to automate some of the validation.
I guess I'm always concerned about using so many snapshots, because it's easy to overlook a real problem when all that's needed is a `jest -u` to update them all 😋 

Not asking for any changes here.
otherwise, this will be a very long test file for such a simple validation 😋